server/server.go

```diff
@@ -165,18 +167,28 @@ func (server *Server) firstTimeInit(configuration *config.Config) error {
 	return nil
 }
 
-func (server *Server) initHttp(configuration *config.Config) {
+func (server *Server) initHttp(configuration *config.Config) error {
 	server.useHttp = true
 	httpPort := configuration.HttpConfig.Port
+	if err := util.CheckPort(httpPort); err != nil {
+		return err
+	}
+
 	server.httpPort = DefaultHttpPort
 	if httpPort != 0 {
```

Comment:
I would remove this check if util.CheckPort( ) requires the port to be > 0 (not >= 0) (which is probably what we want anyway).